models/classes/StatisticalMetadata/Import/Extractor/MetadataHeadersExtractor.php

```diff
@@ -26,8 +26,27 @@
 
 class MetadataHeadersExtractor
 {
+    /** @var array */
+    private $cache = [];
+
     public function extract(array $header): array
     {
-        return preg_grep(sprintf('/^%s/', Header::METADATA_PREFIX), $header);
+        $hash = $this->getHash($header);
+
+        if (!array_key_exists($hash, $this->cache)) {
```

Comment:
See [#r794312964](https://github.com/oat-sa/tao-core/pull/3356/files/74d62b110d8278ddc548d6c1f3196655ff552f54..a420f5f86b20d6006103c4c8e7c3a880c10df26c#r794312964)

```suggestion
        if (!isset($this->cache[$hash])) {
```